app/http.php

```diff
@@ -0,0 +1,366 @@
+
+        // Header used for testing
+        $isProduction = App::getEnv('OPEN_RUNTIMES_PROXY_ENV', 'development') === 'production';
+        if (!$isProduction) {
+            $headers = \array_merge($headers, [
+                'x-open-runtimes-executor-hostname' => $executor['hostname']
+            ]);
+        }
+
+        $client->setHeaders($headers);
+        $client->setData($request->getRawPayload());
+        $client->execute($request->getURI());
+
+        $response
+            ->setStatusCode($client->getStatusCode())
+            ->setContentType(Response::CONTENT_TYPE_JSON, Response::CHARSET_UTF8)
+            ->send($client->getBody());
+    });
+
+// TODO: @Meldiron Uncomment once utopia framework supports wildcard
+/*
+App::error()
+    ->inject('utopia')
+    ->inject('error')
+    ->inject('request')
+    ->inject('response')
+    ->action(function (App $utopia, throwable $error, Request $request, Response $response) {
+        $route = $utopia->match($request);
+        logError($error, "httpError", $route);
+
+        switch ($error->getCode()) {
+            case 400: // Error allowed publicly
+            case 401: // Error allowed publicly
+            case 402: // Error allowed publicly
+            case 403: // Error allowed publicly
+            case 404: // Error allowed publicly
+            case 406: // Error allowed publicly
+            case 409: // Error allowed publicly
+            case 412: // Error allowed publicly
+            case 425: // Error allowed publicly
+            case 429: // Error allowed publicly
+            case 501: // Error allowed publicly
+            case 503: // Error allowed publicly
+                $code = $error->getCode();
+                break;
+            default:
+                $code = 500; // All other errors get the generic 500 server error status code
+        }
+
+        $output = [
+            'message' => $error->getMessage(),
+            'code' => $error->getCode(),
+            'file' => $error->getFile(),
+            'line' => $error->getLine(),
+            'trace' => $error->getTrace(),
+            'version' => App::getEnv('OPEN_RUNTIMES_PROXY_VERSION', 'UNKNOWN')
+        ];
+
+        $response
+            ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+            ->addHeader('Expires', '0')
+            ->addHeader('Pragma', 'no-cache')
+            ->setStatusCode($code);
+
+        $response->json($output);
+    });
+*/
+
+/** @phpstan-ignore-next-line */
+Co\run(
+    function () use ($register) {
+        // Keep updating executors state
+        if (App::getEnv('OPEN_RUNTIMES_PROXY_HEALTHCHECK', 'enabled') === 'enabled') {
+            Timer::tick(\intval(App::getEnv('OPEN_RUNTIMES_PROXY_PING_INTERVAL', '10000')), function (int $timerId) {
+                \go(function () {
+                    fetchExecutorsState(false);
+                });
+            });
+        }
+
+        App::setMode(App::MODE_TYPE_PRODUCTION);
```

Comment:
See previous comment about application mode.

Reply:
Applied as described in previous comment